src/tools/etherscanClient.ts

```diff
@@ -613,34 +630,158 @@ export class EtherscanClient {
     return Number(blockNumberStr);
   }
 
+  private async getLatestBlockNumberAsync(chainId: number): Promise<number> {
+    const blockNumberHex = await this._retrySendAndValidateAsync(
+      {
+        chainId,
+        module: "proxy",
+        action: "eth_blockNumber",
+      },
+      (result) => {
+        if (typeof result !== "string" || !result.startsWith("0x")) {
+          throw new Error(`Invalid block number hex: ${result}`);
+        }
+      },
+    );
+
+    const blockNumber = parseInt(blockNumberHex, 16);
+    log.debug(`[EtherScan] Got latest block number: ${blockNumber}`);
+    return blockNumber;
+  }
+
   public async isBlockIndexedAsync(chainId: number, blockNumber: number): Promise<boolean> {
+    const SAFETY_BUFFER = 20;
+
+    // For Plasma we have no reliable way to check if a block is indexed
+    // Use latestBlock - safetyBuffer approach instead
+    if (chainId === ChainId.Plasma) {
+      try {
+        const latestBlock = await this.getLatestBlockNumberAsync(chainId);
+        const safeBlock = latestBlock - SAFETY_BUFFER;
+        const isIndexed = blockNumber <= safeBlock;
+        
+        return isIndexed;
+      } catch (error) {
+        log.error(`[EtherScan] Error checking if block ${blockNumber} is indexed on Plasma: ${error}`);
+        throw error;
+      }
+    }
+
+    // For other chains, use the original method with eth_getBlockByNumber
```

Comment:
Not really. The original method is `getblockreward`, which is an etherscan method, instead of ethereum JSON RPC method.